sound/soc/sof/sof-client.c

```diff
@@ -68,8 +68,8 @@ int sof_client_dev_register(struct snd_sof_dev *sdev, const char *name, struct i
 	int ret;
 
 	cdev = sof_client_dev_alloc(sdev, name, client_ida);
-	if (IS_ERR(cdev))
-		return PTR_ERR(cdev);
+	if (!cdev)
+		return -ENODEV;
```

Comment:
this looks like a squash in the wrong patch? sof-client.c changes should appear earlier?

Reply:
ugh sorry about that. looks like I messed up the squash!